src/main/clojure/clara/rules/engine.cljc

```diff
@@ -329,6 +329,7 @@
 (defrecord AlphaNode [env children activation]
   IAlphaActivate
   (alpha-activate [node facts memory transport listener]
+    (l/alpha-activate! listener node facts)
```

Comment:
I think you actually want to use the facts that make it through the activation function.  If you look at https://github.com/rbrush/clara-rules/blob/0.12.0/src/main/clojure/clara/rules/compiler.clj#L204 which is where I believe the activation ultimately comes from, you'll see that it can return nil.  The existing logic filters out facts where the activation returns nil from being passed downstream:

````
(for [fact facts
           :let [bindings (activation fact env)] :when bindings] ; FIXME: add env.
       (->Element fact bindings))
````

So, for example, if you have something like

````
(defrule hot-temperatures
      [Temperature (> temperature 100)]
      ;; other-conditions
    => ... )
````

Temperatures that aren't over 100 can just be filtered out in the alpha node.  Right now you're calling the listener on all insertions whose type matched some alpha node, not all insertions that were passed downstream from an alpha node.